messaging/src/main/java/org/axonframework/eventhandling/replay/ReplayAwareMessageHandlerWrapper.java

```diff
@@ -70,5 +76,14 @@ public Object handle(@Nonnull Message<?> message, T target) throws Exception {
             }
             return super.handle(message, target);
         }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <R> Optional<R> attribute(String attributeKey) {
+            if (HandlerAttributes.ALLOW_REPLAY.equals(attributeKey)) {
```

Comment:
If I recall correctly, the `ALLOW_REPLAY` attributes references the `@AllowReplay#value` field, which can _either_ be `true` or `false`.

This depends on whether the users sets an `@AllowReplay` or `@DisallowReplay` annotation, for example.
Or, when somebody sets `@AllowReplay(value = false)` on their event handling component or event handler.

Hence, we are not certain of the decision yet when the `attributeKey` matches, so fixing to "no replay" when the key equals isn't correct I am afraid.

Reply:
By the time we get to this piece of code, the assessment has already been done that `allowReplay==false`. The only thing we need to do here is to check if the attribute "allowReplay" is being asked, in which case we know the answer: `false`. If the question is anything else, we delegate it to the original (wrapped) handler.

Reply:
Taking a second, better look at the implementation, I follow your train of thought. Thanks for the clarification!